kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-kafka-sample/src/test/java/org/kie/server/springboot/samples/KafkaEmitterBrokerDownTest.java

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+
+package org.kie.server.springboot.samples;
+
+import static java.util.Collections.singletonMap;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Optional;
+
+import org.apache.kafka.common.errors.TimeoutException;
+import org.jbpm.casemgmt.api.CaseService;
+import org.jbpm.services.api.DeploymentService;
+import org.jbpm.services.api.ProcessService;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = {KieServerApplication.class, TestAutoConfiguration.class}, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(locations="classpath:application-test.properties")
+@DirtiesContext(classMode= DirtiesContext.ClassMode.AFTER_CLASS)
+public class KafkaEmitterBrokerDownTest extends KafkaFixture {
+    
+    @Autowired
+    protected DeploymentService deploymentService;
+
+    @Autowired
+    protected ProcessService processService;
+    
+    @Autowired
+    protected CaseService caseService;
+    
+    protected String deploymentId;
+
+    @BeforeClass
+    public static void beforeClass() {
+        generalSetup(false);
+        System.setProperty("org.kie.jbpm.event.emitters.kafka.max.block.ms", "500");
+    }
+
+    @After
+    public void cleanup() {
+        cleanup(deploymentService);
+        System.clearProperty("org.kie.jbpm.event.emitters.kafka.max.block.ms");
+    }
+    
+    @Test(timeout = 30000)
+    public void testEmitterNoKafkaBrokerTimeoutException() throws Exception {
+        deploymentId = setup(deploymentService, EVALUATION_PROCESS_ID);
+        
+        ListAppender<ILoggingEvent> listAppender = addLogAppender();
+        
+        Long processInstanceId = processService.startProcess(deploymentId, EVALUATION_PROCESS_ID, singletonMap("initiator", YODA));
+
+        Optional<ILoggingEvent> logEvent = getLog(listAppender);
+        assertEquals(TimeoutException.class.getCanonicalName(), logEvent.get().getThrowableProxy().getClassName());
+        
+        assertNotNull(processInstanceId);
+        assertTrue(processInstanceId > 0);
+
+        processService.abortProcessInstance(processInstanceId);
+        
+        ProcessInstance pi = processService.getProcessInstance(processInstanceId);
+        assertNull(pi);
```

Comment:
wouldn't it be worth checking that KafkaEmitter is able to send messages to the topic once the timeout is removed (or increased)? - wdyt?

Reply:
Not in this test, as there's no testcontainer kafka broker